crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -45,43 +40,17 @@ pub(crate) fn do_finalize_domain_current_epoch<T: Config>(
     // Reset pending staking operation count to 0
     PendingStakingOperationCount::<T>::set(domain_id, 0);
 
-    // slash the operators
-    do_finalize_slashed_operators::<T>(domain_id).map_err(Error::SlashOperator)?;
-
     // re stake operator's tax from the rewards
     operator_take_reward_tax_and_stake::<T>(domain_id)?;
 
+    // slash the operators
```

Comment:
Why is slashing moved here?

Reply:
in the previous version, we were slashing first and then distributing rewards for the operators after. While distributing, if the operator is not found, then reward is not given. Operator cannot be found if they were slashed in the previous step. So treasury was essentially not getting the rewards for the last epoch. Moving slashing after the reward distribution ensures last epoch rewards are included in the slashed amount

Reply:
By treasury you mean the escrow? That takes % of block storage fees? If so that is going to be deprecated. Can't we share the fees among the rest of non-slashed operators who submitted this ER? Anyway I think this shouldn't be discussed in this PR then as it's unrelated to new staking TBH.

Reply:
Nope, we did not have any escrow at the time of staking v1. So instead of burning SSC that is left over, we ended giving them to a treasury account. We can still burn them as well. 

I'll leave this as is for now until we make a decision on who get that funds. Change should be trivial anyway